### PR TITLE
Update mmap stubs for newer Python versions

### PR DESCRIPTION
**Pull request by bmerry**
Based on the Python stdlib documentation:
- Since Python 3.5, mmap.{find,rfind,write} all accept any bytes-like.
  I've used the _typeshed.ReadableBuffer alias defined in #4232.
- Since Python 3.6, mmap.write returns the number of bytes written.
- Since Python 3.3, mmap.read allows None as the parameter; while in
  Python 2 the argument cannot be omitted.

### CODE REVIEW

**Comment by bmerry** [comment]
Incidentally, I don't really follow why master defines `rfind` separately for Python 2/3 while it defines `find` in the base class. I'd be happy to clean that up too if there is no good reason for it.

File: stdlib/2and3/mmap.pyi [hunk 1]
```diff
@@ -56,7 +63,15 @@ if sys.version_info >= (3,):
         closed: bool
         if sys.version_info >= (3, 8) and sys.platform != "win32":
             def madvise(self, option: int, start: int = ..., length: int = ...) -> None: ...
-        def rfind(self, sub: bytes, start: int = ..., stop: int = ...) -> int: ...
+        if sys.version_info >= (3, 5):
+            def find(self, sub: ReadableBuffer, start: int = ..., stop: int = ...) -> int: ...
+            def rfind(self, sub: ReadableBuffer, start: int = ..., stop: int = ...) -> int: ...
+        else:
+            def rfind(self, sub: bytes, start: int = ..., stop: int = ...) -> int: ...
+        if sys.version_info >= (3, 6):
```
Comment (srittau) [inline stdlib/2and3/mmap.pyi:71]: Since typeshed only supports Python 2.7 and 3.5+, you can leave out the version check and the else branch.

Comment (srittau) [reply stdlib/2and3/mmap.pyi:71]: And the mark is wrong, this is for lines 66 to 70.

File: stdlib/2and3/mmap.pyi [hunk 2]
```diff
@@ -56,7 +63,15 @@ if sys.version_info >= (3,):
         closed: bool
         if sys.version_info >= (3, 8) and sys.platform != "win32":
             def madvise(self, option: int, start: int = ..., length: int = ...) -> None: ...
-        def rfind(self, sub: bytes, start: int = ..., stop: int = ...) -> int: ...
+        if sys.version_info >= (3, 5):
+            def find(self, sub: ReadableBuffer, start: int = ..., stop: int = ...) -> int: ...
+            def rfind(self, sub: ReadableBuffer, start: int = ..., stop: int = ...) -> int: ...
+        else:
+            def rfind(self, sub: bytes, start: int = ..., stop: int = ...) -> int: ...
```
Comment (srittau) [inline stdlib/2and3/mmap.pyi:70]: Since typeshed only supports Python 2.7 and 3.5+, you can leave out the version check and the else branch.

**Comment by srittau** [comment]
Generally, some type stubs are rather old and use conventions that predate modern typeshed conventions. They also often have been patched several times by multiple people. So if you find something amiss, it's often due to these factors, and cleanups are most welcome. Checking the output of `git log` and `git blame` for the file in question can also give hints.

File: stdlib/2and3/mmap.pyi (hunk 2 above)
Comment (bmerry) [reply stdlib/2and3/mmap.pyi]: Thanks, I've used that to simplify/clean up things a bit further.

**Comment by bmerry** [comment]
> I'm wondering what's going on with the "_mmap" class.

Glad it's not just me that's confused. I'm pretty tempted to just reorganise it as a single class with `if sys_version` blocks.

**Pull request merged**